src/compiler/checker.ts

```diff
@@ -13483,7 +13483,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             for (const current of type.types) {
                 for (const prop of getPropertiesOfType(current)) {
                     if (!members.has(prop.escapedName)) {
-                        const combinedProp = getPropertyOfUnionOrIntersectionType(type, prop.escapedName);
+                        const combinedProp = getPropertyOfUnionOrIntersectionType(type, prop.escapedName, /*skipObjectFunctionPropertyAugment*/ !!(type.flags & TypeFlags.Intersection));

```

Comment:
why is this unconditional here but has a fallback in the `getPropertyOfType` code?

Reply:
This resolves "own" members of the intersection so we never want to pull from function property augment here.

The other place is responsible for getting a `name` out of all accessible properties, it prioritizes its own ones but has to fallback to the ones from the augment to account for this:
```ts
interface Test1 { toString: null | 'string'; }
type Test2 = Test1 & { optional?: unknown };
declare const source: Test1;
const target: Test2 = { ...source };

const toString = target.toString; // own one, augment is avoided here
const hasOwn = target.hasOwnProperty; // not an own member but it should still be accessible
```

I pushed out this (`hasOwn` bit) as an extra thing in the added test case.